master/master.cpp

```diff
@@ -63,6 +61,10 @@ Master::Master() {
     lb_step = -2;
     partition_to_donate.resize(partition_table.size(), false);
     debug_num_broadcast = 0;
+
+    for(auto setup_handler : external_setup_handlers) {
```

Comment:
@Yuzhen11 These are std::function, so using reference won't make it more efficient? We can verify it anyway.